### PR TITLE
Add support for kubernetes backend for ESO

### PR DESCRIPTION
**Pull request by mhjacks**

### CODE REVIEW

File: ansible/roles/vault_utils/values-secrets.v2.schema.json [hunk 1]
```diff
@@ -94,7 +94,7 @@
         "version": {
           "type": [ "string", "null" ],
           "description": "Version of the secret specification",
-          "default": "1.0"
+          "default": "2.0"
```
[the remark this answers is not in the document]
Comment (mhjacks) [reply ansible/roles/vault_utils/values-secrets.v2.schema.json]: Fair enough. Reverted.

File: Makefile [hunk 2]
```diff
@@ -77,8 +77,32 @@ uninstall: ## runs helm uninstall
...
+load-secrets: ## loads the secrets into the backend determined by values-secret config file
+	common/scripts/process-secrets.sh $(NAME)
+
+.PHONY: secrets-backend-vault
+secrets-backend-vault: ## Edits values files to use default Vault+ESO secrets config
+	common/scripts/set-secret-backend.sh vault
+	common/scripts/manage-secret-app.sh vault present
+	common/scripts/manage-secret-app.sh golang-external-secrets present
+	common/scripts/manage-secret-namespace.sh validated-patterns-secrets absent
+	@echo "Secrets backend set to vault, please review changes, commit, and push to activate in the pattern"
```
[the remark this answers is not in the document]
Comment (mhjacks) [reply Makefile]: Added.

**Comment by mhjacks** [comment]
Rebased onto latest main as of now. Latest commits address a potential issue with retrieving embedded newlines from a YAML file - both the policy retrieval code and the value retrieval code in `parse_secrets_info` now handle this situation; and there are tests in `ansible-unittests` that test for correct newline handling.

**Review by mbaldessari — APPROVED** [review]
/lgtm

**Pull request merged**